### PR TITLE
Refactor gci and cnudie

### PR DESCRIPTION
**Pull request by ccwienk**

### CODE REVIEW

File: gci/oci.py [hunk 1]
```diff
@@ -106,34 +106,3 @@ def component_descriptor_from_tarfileobj(
           raise ValueError('Component Descriptor appears to be empty')
 
         return ocm.ComponentDescriptor.from_dict(raw_dict)
-
-
-def image_ref_with_digest(
```
Comment (8R0WNI3) [inline gci/oci.py:111]: Actually, this function is being used [here](https://github.com/open-component-model/delivery-service/blob/cf8ea385c6b6940c0bc0a2ad95313228c8f5d515/bdba/util.py#L299) in delivery-service. That being said, I'm not against dropping this function (here), but we'll have to adjust delivery-service accordingly first.

File: cnudie/upload.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 import cnudie.util
 import ocm
-import gci.oci
+import ocm.oci
```
Comment (8R0WNI3) [inline cnudie/upload.py:15]: nit: Please order imports alphabetically.

File: ctt/replicate.py [hunk 3]
```diff
@@ -8,11 +8,10 @@
 import json
 import logging
 
-import gci.oci
-
 import ccc.oci
-import ocm
 import oci
+import ocm
+import ocm.oci
```
Comment (8R0WNI3) [inline ctt/replicate.py:14]: ditto

File: gci/oci.py (hunk 1 above)
Comment (ccwienk) [reply gci/oci.py:111]: indeed. I think moving this function to delivery-service will make sense. maybe we can drop some codepaths

**Review by 8R0WNI3 — APPROVED** [review]
/lgtm

**Pull request merged**